lib/rbac/filterer.rb

```diff
@@ -367,7 +380,7 @@ def search(options = {})
       end
 
       # Preserve sort order of incoming target_ids
-      if !target_ids.nil? && !order
+      if !target_ids.nil? && !order && !targets.kind_of?(Array)
```

Comment:
please share reason for this change.
The implications of this change will impact many many places of the application.

I do agree, that this sorting is called in a few places that I wish it were not.

The solution I find is to pass in a scope and that solves this problem.